Source/CarthageKit/BinaryProject.swift

```diff
@@ -67,4 +69,12 @@ public struct BinaryProject: Equatable {
 				return .success(BinaryProject(versions: versions))
 			}
 	}
+    
+    public static func isValidBinaryUrl(value: String) -> Bool {
+        let detector = try! NSDataDetector(types: NSTextCheckingResult.CheckingType.link.rawValue)
+        let matches = detector.matches(in: value, options: [], range: NSRange(location: 0, length: value.utf16.count))
+        let match = matches.first(where: { Range($0.range, in: value) != nil })
+        
+        return match != nil
+    }
```

Comment:
I think I'd prefer we use the same URL and scheme check that this function uses elsewhere. As written, this will accept a parameter like `?alt=foo.co`, which will just be rejected later on as a `.nonHTTPSURL`.

What would you think about doing something like this?
```suggestion
	private static func validateBinaryURL(_ value: String) -> Result<URL, BinaryJSONError> {
		guard let binaryURL = URL(string: value) else {
			return .failure(BinaryJSONError.invalidURL(value))
		}
		guard binaryURL.scheme == "file" || binaryURL.scheme == "https" else {
			return .failure(BinaryJSONError.nonHTTPSURL(binaryURL))
		}
		return .success(binaryURL)
	}
```


and then checking further up, like
```swift
else if item.name == "alt", let value = item.value, 
	 case .success = validateBinaryURL(value) {
	// ...
}
```

We could also refactor `from(jsonData:)` to use this function instead of repeating the checks.